### PR TITLE
#163519144 Build API end-point for articles (CRUD)

### PR DESCRIPTION
**Pull request by filschristian**
#### What does this PR do?
Have CRUD API end-points for articles

#### Description of Task to be completed?
Have the following end-point working:
`POST api/v1/articles`
`GET api/v1/articles/:slug`
`PUT api/v1/articles/:slug`
`DELETE api/v1/articles/:slug`
`GET api/v1/articles?tag=&author=&favorited=`

#### How should this be manually tested?

- Download/clone the repo

- run `npm install`

- run `npm run migrate:seed`

- run `npm run test`

#### Any background context you want to provide?
Check the read me for different request and response object format

#### What are the relevant pivotal tracker stories?
#163519144

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by kimotho-njoki** [comment]
@oesukam @filschristian what is going on with tests?

File: config/cloudinary.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  api_secret: CLOUD_API_SECRET
+});
+
+const storage = (folder = 'articles') =>
+  cloudinaryStorage({
+    cloudinary,
+    folder,
+    allowedFormats: ['jpg', 'png'],
+    transformation: [{ width: 500, height: 500, crop: 'limit' }]
+    //   filename(req, file, cb) {
```
Comment (kimotho-njoki) [inline config/cloudinary.js]: were you supposed to push this comment to github?

**Comment by filschristian** [comment]
> @oesukam @filschristian what is going on with tests?

our lines' coverage is less than 80% that's why they are failing, we are going to add other tests

File: middlewares/verifyJwt.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import passport from 'passport';
+
+const verifyJwt = ({ tokenRequired = true, confirmEmail = false } = {}) => {
```
[the remark this answers is not in the document]
Comment (kimotho-njoki) [reply middlewares/verifyJwt.js]: We are still ignoring hound comments huh?

File: config/cloudinary.js (hunk 1 above)
Comment (oesukam) [reply config/cloudinary.js]: Resolved. They have been removed

Comment (oesukam) [reply config/cloudinary.js]: Resolved by deleting those comments

File: middlewares/verifyJwt.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (oesukam) [reply middlewares/verifyJwt.js]: Resolved by removing the body bracket of verifyJwt function

**Pull request merged**